client/my-sites/checklist/wpcom-checklist/checklist-banner/index.jsx

```diff
@@ -65,20 +65,18 @@ export class ChecklistBanner extends Component {
 	}
 
 	render() {
-		const { translate } = this.props;
-
 		if ( ! this.canShow() ) {
 			return null;
 		}
 
+		const { translate, taskList } = this.props;
+		const tasks = taskList.getAll();
+
 		const childrenArray = Children.toArray( this.props.children );
-		const total = childrenArray.length;
-		const completeCount = reduce(
-			childrenArray,
-			( sum, child ) => ( !! child.props.completed ? sum + 1 : sum ),
-			0
-		);
-		const isFinished = completeCount >= total;
+		const total = tasks.length;
+		const completeCount = tasks.filter( task => task.isCompleted ).length;
+		const isFinished = taskList.areAllTasksCompleted();
+		const firstIncomplete = taskList.getFirstIncompleteTask();
 		const completePercentage = Math.round( ! total ? 0 : ( completeCount / total ) * 100 );
```

Comment:
What if there was a `getCompletedPercentage` method on the `taskList`? Or `getCompletionStatus` that would return an object that contains everything our UI wants to know:
```js
{
  completed: 6,
  total: 10,
  percent: 60,
}
```
Just an idea on how to improve the object-oriented API of `WpcomTaskList` 🙂

Reply:
Great idea! 😄